models/DataObject/QuantityValue/UnitConversionService.php

```diff
@@ -51,16 +47,12 @@ public function convert(AbstractQuantityValue $quantityValue, Unit $toUnit)
         if ($baseUnit === null) {
             $baseUnit = $toUnit;
         }
-        $converterServiceName = $baseUnit->getConverter();
 
-        if ($converterServiceName) {
-            $converterService = $this->container->get($converterServiceName);
-        } else {
-            $converterService = $this->container->get(QuantityValueConverterInterface::class);
-        }
+        $converterServiceName = $baseUnit->getConverter();
+        $converterService = UnitConverterResolver::resolveUnitConverter($converterServiceName);
 
         if (!$converterService instanceof QuantityValueConverterInterface) {
-            throw new \Exception('Converter class needs to implement '.QuantityValueConverterInterface::class);
```

Comment:
Why not throw an exception if the service name is wrong? I think the default service should only be used if no service name is set

Reply:
yes, [this](https://github.com/pimcore/pimcore/pull/12742) should already throw exception. maybe we can refactor a bit to only resolve converter when the value is set.

Reply:
@dvesh3 This will only throw an exception if the service doesn't exist. But returns null if only the interface is missing. Right? I think we should hold the logic here as it was before.

```
        $converterServiceName = $baseUnit->getConverter();
        if ($converterServiceName) {
            $converterService = UnitConverterResolver::resolveUnitConverter($converterServiceName);
        } else {
            $converterService = $this->defaultConverter;
        }

        if (!$converterService instanceof QuantityValueConverterInterface) {
            throw new \Exception('Converter class needs to implement '.QuantityValueConverterInterface::class);
        }
```

Reply:
I got your point now. yes, this changes behavior as it silently fallback to the default converter if the custom converter service is not implementing `QuantityValueConverterInterface`. 

Changed it as per suggestion. thanks!